src/nspawn/nspawn.c

```diff
@@ -1096,6 +1096,17 @@ static int parse_argv(int argc, char *argv[]) {
         if (r < 0)
                 return r;
 
+	// When cgroup namespaces are supported the init in the container will
+	// mount cgroups itself. Hence, when we detect that the user requested
+	// that unified cgroups should be mounted or we detect that the host is
+	// using unified we need to pass "systemd.unified_cgroup_hierarchy=1" to
+	// the containers init to prevent it from mounting the current default
+	// legacy cgroup.
+	if (arg_unified_cgroup_hierarchy && cg_ns_supported() && arg_start_mode == START_BOOT) {
+		if (strv_extend(&arg_parameters, "systemd.unified_cgroup_hierarchy=1") < 0)
+			return log_oom();
```

Comment:
@brauner , thanks!
```
systemd-nspawn -D /nspawn-root/ -b 3
```
works fine.
But
```
-bash-4.3# systemd-nspawn -D /nspawn-root/ /usr/lib/systemd/systemd 3
...
container# grep cgroup /proc/self/mounts
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/net_cls cgroup rw,nosuid,nodev,noexec,relatime,net_cls 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
```
This is a regression.

Another issue: we overwrites the user's setting
```
-bash-4.3# systemd-nspawn -D /nspawn-root -b 3 systemd.unified_cgroup_hierarchy=0
...
-bash-4.3# grep cgroup /proc/self/mounts
cgroup /sys/fs/cgroup cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
```
(actually, `systemd.unified_cgroup_hierarchy=...` never really works. So this is not a regression. Maybe, we should document this)